vault.go

```diff
@@ -481,6 +481,39 @@ func newVaultClient(opts *config) (*api.Client, error) {
 	// step: set the token for the client
 	client.SetToken(token)
 
+	if opts.vaultRenewToken {
+		tokeninfo, err := client.Auth().Token().LookupSelf()
+		if err != nil {
+			return nil, fmt.Errorf("failed to lookup token info: %s", err)
+		}
+
+		tokenttl, err := tokeninfo.TokenTTL()
+		if err != nil {
+			return nil, fmt.Errorf("failed to lookup token ttl: %s", err)
+		}
+		glog.Infof("token ttl is %v", tokenttl)
+		renewPeriod := tokenttl / 2
+		go func() {
+			for {
+				glog.Infof("scheduling token renew in %v", renewPeriod)
+				<-time.After(renewPeriod)
+				glog.Infof("attempting to renew token")
+				newtokeninfo, err := client.Auth().Token().RenewSelf(0)
+				if err != nil {
+					renewPeriod = renewPeriod / 2
```

Comment:
I guess an issue with this is, assuming a few errors or service unavailable, you can negative pretty quickly and start to eat cycles, perhaps minimum or some former of backoff

Reply:
This is true. 

Putting a backoff in makes sense I guess, but it is kind of weird to continue trying after the ttl of the secret is expired. 

Maybe we should just exit 1 when the renewPeriod gets below a minimum. What do you think?

Reply:
Hi @roboll, sorry for delay ... yes, agreed .. I'd rather exit 1 on a negative ttl and allow kubernetes to do the backoff (hopefully the user should have monitoring in place for the crashloopbackoff) and perhaps exchange the / 2 for a either a [backoff](https://github.com/jpillora/backoff) or static time.After? ... what you reckon?